wit/gpio.wit

```diff
@@ -0,0 +1,38 @@
+variant gpio-error {
+  configuration-error(string),
+  pin-usage-error(string),
+  hardware-error(string),
+  unexpected-error(string),
+}
+
+enum logic-level {
+  low,
+  high,
+}
+
+resource input-pin {
+  static get-named: func(name: string) -> expected<input-pin, gpio-error>
+
+  read: func() -> logic-level
+}
+
+resource output-pin {
+  static get-named: func(name: string) -> expected<output-pin, gpio-error>
+
+  write: func(level: logic-level) -> unit
+}
+
+/// A pin that can output a PWM signal.
+resource pwm-output-pin {
+  /// Acquire a handle to the pin named in the slightfile.
+  static get-named: func(name: string) -> expected<pwm-output-pin, gpio-error>
+
+  /// Configure the pin's duty cycle, which should be and will be clamped between 0.0 and 1.0.
+  /// This does not enable the pin if it is currently disabled.
+  set-duty-cycle: func(duty-cycle: float32) -> unit
+
+  /// Enable the output signal.
+  enable: func() -> unit
+  /// Disable the output signal.
+  disable: func() -> unit
+}
```

Comment:
Usually, we'd first get a proposal for an interface, and then later move on to the implementation to allow for agreement over the interface prior to any work being done – That said, I think this is looking very solid. It'd be cool to be able to add some sort of pin event handling to be able to register callbacks or smt based on specific GPIO pin events, but callback support in WIT is very restricted.